### ISSUE TITLE
Race condition bug on "my-fueled-app" example

### ISSUE THREAD

**Issue by itspauloroberto** [FuelLabs/fuels-ts#499]
### What version of fuels-ts are you using?

Latest (v0.13.0)

### Steps to Reproduce

1. Run the "my-fueled-app" sample application according to the Quickstart guidelines walkthrough.
2. Click the "Increment" button several times, and the faster you can. (You can also use a script to click on it very fast to make it easier to reproduce)
3. Look into the console of the fuel-core API running, and you can see errors mentioning: "Transaction id was already used".

### Expected Behavior

The expected behavior is to use the cleanup function to prevent newer promises from being fired before older ones get fulfilled.

Or it can also be a loading state, or even to keep track of the last used transaction id, but I do feel that's too much for a simple task.

### Actual Behavior

Basically, the errors are popping up on the console of the fuel-core API running. And this is because of the promise race condition bug which occurs because the old promise was fulfilled before the newer promise. The application is not handling that edge case because there is no cleanup being used on hooks or there is no loading state or something like this.

**Comment by digorithm** [comment]
@luizstacio what's the current status on this?
